### PR TITLE
fstar-debug-invocation: make sure to start F* in the same directory e…

### LINKED ISSUE #93: fstar-debug-invocation and include paths
I have the following saucy invocation:
```
emacs -f fstar-debug-invocation /home/guido/r/fstar/bin/fstar.exe --max_fuel 1 --max_ifuel 1 --initial_ifuel 0 --z3cliopt smt.arith.nl=false --z3cliopt smt.QI.EAGER_THRESHOLD=100 --z3cliopt smt.CASE_SPLIT=3 --hint_info --use_hints --cache_checked_modules --use_extracted_interfaces true --smtencoding.elim_box true --smtencoding.l_arith_repr native --smtencoding.nl_arith_repr wrapped --smt /home/guido/bin/z3/z3-4.5.1.1f29cebd4df6-x64-ubuntu-14.04/bin/z3 --include obj/Vale/test --include obj/test --include obj/arch --include obj/arch/x64/ --include obj/lib/collections/ --include obj/lib/math --include obj/lib/util --include obj/crypto/aes/ --include obj/crypto/aes/x64 --include obj/crypto/poly1305/ --include obj/crypto/poly1305/x64/ --include obj/thirdPartyPorts/OpenSSL/poly1305/x64/ --include obj/external/ obj/crypto/aes/x64/FastMul_helpers_i.fst
```
It opens `FastMul_helpers_i.fst` correctly, and runs F* with the options exactly as shown, but it seems F* is running inside the `obj/crypto/aes/x64` directory, since it fails with:
```
not a valid include directory: /home/guido/r/everest/vale/obj/crypto/aes/x64/obj/Vale/test
```
when it should really be pointing to `/home/guido/r/everest/vale/obj/Vale/test`. FWIW I think it's just plain emacs that's changing the directory, but could we work around it?

### PR DESCRIPTION
**Pull request by mtzguido**
…macs started

This is in order to properly interpret paths.

Fixes #93.

### CODE REVIEW

File: fstar-mode.el [hunk 1]
```diff
@@ -5225,9 +5226,12 @@ Current file: %S" fstar-executable fstar-subp-prover-args buffer-file-name)
 
 (defconst fstar-subp--process-name "F* interactive")
 
+(defvar fstar-subp--default-directory nil)
```
Comment (cpitclaudel) [inline fstar-mode.el]: Can you add documentation for this variable? :)

Comment (mtzguido) [reply fstar-mode.el]: Sure! But you got me thinking, why don't we just do this?
```diff
diff --git a/fstar-mode.el b/fstar-mode.el
index 4580391..6214fe8 100755
--- a/fstar-mode.el
+++ b/fstar-mode.el
@@ -5227,7 +5227,8 @@ Current file: %S" fstar-executable fstar-subp-prover-args buffer-file-name)
 
 (defun fstar-subp--start-process (buf prog args)
   "Start an F* subprocess PROG in BUF with ARGS."
-  (apply #'start-file-process fstar-subp--process-name buf prog args))
+  (let ((default-directory command-line-default-directory))
+    (apply #'start-file-process fstar-subp--process-name buf prog args)))
 
 (defun fstar-subp--wait-for-features-list (proc)
   "Busy-wait until the first protocol-info message from PROC."
```
Does `command-line-default-directory` stay constant throughout a session? In any case if there's a reason to prefer the other solution I'll add documentation and update the PR, no strong feeling of course.

Comment (cpitclaudel) [reply fstar-mode.el]: I don't think that will work: if I open emacs in `~/` then open `~/git/project/xyz.fst`, I don't want F* to start in `~/`.

Comment (mtzguido) [reply fstar-mode.el]: That's right, I think I would actually prefer that myself :). But sorry, I'm bikesheding, so let's keep this and I'll add documentation.

**Pull request merged**